File: examples/javascript/proof.js
import axios from "axios";
import path from "path";
import { fileURLToPath } from "url";

const API_URL = 'http://localhost:3030';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function generateElf() {
    const url = `${API_URL}/sp1/generate-elf`;

    const destinationPath = path.join(__dirname, "..", "elf");

    const data = { destination_path: destinationPath };

    return axios.post(url, data)
        .then(response => response.data)
        .catch(error => {
            console.error('Error:', error);
            throw error;
        });
}

(async () => {
    try {
        const result = await generateElf();
        console.log(`Result: ${JSON.stringify(result)}`);
    } catch (error) {
        console.error('Failed to generate ELF:', error);
    }
})();

File: examples/python/proof.py
import os
import requests

API_URL = "http://localhost:3030"

def generate_elf():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    destination_path = os.path.join(script_dir, "examples", "elf")
    
    url = f"{API_URL}/sp1/generate-elf"
    data = {"destination_path": destination_path}
    
    response = requests.post(url, json=data)
    return response.json()

if __name__ == "__main__":
    result = generate_elf()

    print(f"Result: {result}")
